Opentrons/opentrons-emulation
# test: Add tests for input files

# Overview

Add tests for input files. Add some properties that were created during creation of tests

# Changelog

- configuration_file.py
  - Add check to confirm `robot` and `modules` items exist for `validate_names` validator
  - Add `there_can_only_be_one` validator to enforce there being only a single robot in a configuration 
  - Add `modules_exist` and `robot_exists` boolean properties
  - Add `get_by_id` method to load a container by it's name
  - Add `from_file` and `from_dict` constructor methods
  - Remove run function
- Module and robot tests
  - Test emulation levels and source repos
- SystemConfigurationModel tests
  - Add test to confirm container name format is being parsed correctly
  - Add test to confirm ValidationError is thrown when robot and module have the same name
  - Add test to confirm ValidationError is thrown when there are multiple robots
  - Add test to confirm functionality of `modules_exist` and `robot_exists` property
  - Add test to confirm `containers` property
  - Add test to confirm robot and modules are loaded correctly when using `get_by_id` method
  - Add test to confirm `from_file` method is working

# Review requests

None

# Risk assessment

None, adding tests. I would expect this to make things more stable instead of less stable. Unless I suck at my job, which is a definite possibility.
